src/modules/auth/services/auth.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { JwtService } from '@nestjs/jwt';
+import { UsersService } from '../../users/interfaces/UsersService';
+
+import type { CreateUserDto } from '../../users/dtos/createUser.dto';
+import type { AuthDto } from '../dtos/auth.dto';
+import type { AuthService } from '../interfaces/AuthService';
+import type { Tokens } from '../types/tokens.type';
+
+@Injectable()
+export class AuthServiceImpl implements AuthService {
+
+  constructor(
+    private usersService: UsersService,
+    private jwtService: JwtService,
+    private configService: ConfigService,
+  ) {}
+
+  public async signin(dto: AuthDto): Promise<Tokens> {
+    const user = await this.usersService.validateUser(dto);
+
+    const tokens = await this.getTokens(user.id, user.email);
+    this.updateToken(user.id, tokens.refresh_token);
+
+    return tokens;
+  }
+
+  public async signup(dto: CreateUserDto): Promise<Tokens> {
+    const newUser = await this.usersService.createUser(dto);
+
+    const tokens = await this.getTokens(newUser.id, newUser.email);
+    this.updateToken(newUser.id, tokens.refresh_token);
+
+    return tokens;
+  }
+
+  public async logout(id: number): Promise<void> {
+    await this.usersService.updateToken(id, null);
+  }
+
+  public async refresh(id: number, rt: string): Promise<Tokens> {
+    const user = await this.usersService.validateToken(id, rt);
+
+    const tokens = await this.getTokens(user.id, user.email);
+    this.updateToken(user.id, tokens.refresh_token);
+
+    return tokens;
+  }
+
+  public async updateToken(id: number, rt: string): Promise<void> {
+    await this.usersService.updateToken(id, rt);
+  }
+
+  public async getTokens(id: number, email: string): Promise<Tokens> {
```

Comment:
Якщо цей метод використовується лише у цьому сервісі тоді вартує зробити його приватним, також видалити його з абстрактного класу, це стосується всіх сервісів
```suggestion
  private async _getTokens(id: number, email: string): Promise<Tokens> {
```